backend/fms_core/management/commands/_delete_individual.py

```diff
@@ -0,0 +1,86 @@
+from django.apps import apps
+import reversion
+import json
+import logging
+from reversion.models import Version
+from fms_core.models import Individual
+
+# Parameters required for this curation
+ACTION = "action"                     # = delete_individual
+CURATION_INDEX = "curation_index"     # Number indicating the order in which this action was performed during the curation.
+COMMENT = "comment"                   # An optional comment to be stored in the logs
+INDIVIDUAL_NAMES = "individual_names" # An array of individual name that are to be deleted.
+USER_ID = "requester_user_id"         # The user id of the person requesting the curation. Optional. If left empty, uses biobankadmin id.
+
+# Curation params template
+# { CURATION_INDEX: 1,
+#   ACTION: "delete_individual",
+#   COMMENT: "Dr. No asked to delete these individuals that are not used.",
+#   INDIVIDUAL_NAMES: ["Roger", "TiPaul", "Gertrude"], # List of individual names (unique) to be deleted
+#   USER_ID: 5
+# }
+
+# This curation deletes only individual that are not referenced. Remove individual from samples references using "update_field_value" first.
+# To delete parents, a first curation need to be run to delete the child individual (or an update), then parents will be unreferenced.
+
+# function that checks the references to an individual and list them.
+def references(individual):
+    links = []
```

Comment:
Can you use a more concise logic? reassigning the children variable and rechecking it for both children as mother and father does not look straightforward to me. 

Also, I'd make that a model method that lists the children samples, and another method that collects the samples from the collection of X individuals.

Reply:
I hardly see this as a useful model method. This is very specific to this curation needs (the fact that it return human readable references for the logs). Maybe we could think of an alternate version that returns a list in a less human readable form for a method. 
As for being concise, i could save a couple lines by using different variables. The idea here is to give a useful log information. Separating mother and father helps to identify the error. Not sure i can save much there ...

Reply:
Just a suggestion, but something like this will be much more performant querywise and memory-wise, especially if we are to have lot of individuals (millions in the future?)

    reference_logs = []
    if individual.mother_of.count() > 0:
        reference_logs.append(f"Mother of {list(individual.mother_of.values_list('id', flat=True))}.")

Reply:
Not sure this would save much of anything ... unless we have very fertile mothers ... ;)

Reply:
right.. lol an individual should not have many children individuals or many samples... ok.. lol

Reply:
It must admit i am not very up to date with the performance tweaking of those inline string builders. The way python convert that string list into a single string may be ineffective, but it might not be worth using less readable code for those curations that will always stay within a reasonable scope. If we have a blanket curation to do on all samples or individuals we'd use a migration or at least design a different script (one that would not list each individual to delete, but instead the condition that need to be met for deletion).